### PR TITLE
Add icons

### LINKED ISSUE #123: Mike icons migration
Related to migrating the mike icons into the react-components lab and the considerations in this regard such as using the icons from a separate package and so on.

CC @rihr

### PR DESCRIPTION
**Pull request by rihr**
Adds icons package along with all MIKE icons
Started in and closes #149
Closes #123

### CODE REVIEW

**Comment by rihr** [comment]
@bertearazvan @rosudhi I thought it would be better to make a new branch so we can more easily compare the two

This implementation is meant to be very simple:
- in the src dir: only SVG files - the file names have to be in `kebab-case` and the SVGs must be 40x40px in size (all MIKE icons are 40px as far as I know)
- the `build-icons.js` file is run with `yarn build`. It generates TSX format components from an SVGR template, then they are compiled with typescript into js and d.ts files.
- Icons can then be imported directly like this `import Edit from '@dhi/icons/Edit'`

File: packages/icons/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+<div align="left">
```
Comment (bertearazvan) [inline packages/icons/README.md:1]: Probably we can remove this, or at least some parts of it.

**Review by bertearazvan — APPROVED** [review]
Looking good and I really like the whole setup

**Comment by bertearazvan** [comment]
I am adding the pipeline with semantic-release :)

File: packages/icons/README.md (hunk 1 above)
Comment (rihr) [reply packages/icons/README.md:1]: Yeah, just need something about what size, name, etc the svgs have to before they can be added

**Comment by bertearazvan** [comment]
It would be nice if we could also import them like `import { Icon1, Icon2 } from "@dhi/icons"`

**Comment by rihr** [comment]
> It would be nice if we could also import them like `import { Icon1, Icon2 } from "@dhi/icons"`

I can think of a hacky but easy way of adding it, will try

**Comment by rihr** [comment]
Haven't tested but should work now

**Comment by rihr** [comment]
Confirmed actually working now 😄

**Comment by bertearazvan** [comment]
I believe that the `files` field is required for adding those files when publishing the package?

**Comment by rosudhi** [comment]
looking good, just see some typo in MD files related to react-components-icons text

**Comment by rihr** [comment]
> I believe that the `files` field is required for adding those files when publishing the package?

https://docs.npmjs.com/cli/v7/configuring-npm/package-json#files It's optional, otherwise everything is published

**Comment by bertearazvan** [comment]
> > I believe that the `files` field is required for adding those files when publishing the package?
> 
> https://docs.npmjs.com/cli/v7/configuring-npm/package-json#files It's optional, otherwise everything is published

So in this case we also want the src to be published? Wouldn't the dist be enough for using them? Always wondered, thought the src is simply in case you want to go to the node_modules and look the components up, or smth

**Comment by rihr** [comment]
Nope, that's not the same package.json file
See here https://github.com/DHI/react-components/pull/156/files#diff-86dab1fe01eb938c9d60e7d3a98590a87859ba326fb4eb18b9a4a7258c98f0ccR71
It's being copied and modified inside `dist`
Not sure if there's actually much benefit to doing that - it would work fine with just the single file

**Comment by bertearazvan** [comment]
I'll write the README part today. Then I think we are ready to publish! 

Maybe we should add more icons to it too?

**Comment by rosudhi** [comment]
> I'll write the README part today. Then I think we are ready to publish!
> 
> Maybe we should add more icons to it too?

to make it quick, i think we can just merge it now, later we can add more icons and improvement.

**Comment by rihr** [comment]
It shouldn't take too long to add in the icons, I can add them

**Comment by bertearazvan** [comment]
I would write the README part before merging, so we have a good initial state of the package.

**Comment by bertearazvan** [comment]
@rihr Feel free to review the Readme, if there are any typos or anything that needs rewording.

**Comment by rihr** [comment]
Apart from making sure auto versioning and release works, I think it's ready for merge now.
I just made some smaller edits to the contrib doc, so make sure that's all good

**Review by bertearazvan — APPROVED** [review]
Super nice edits in the readme and thanks for using primarily one package.json. Super nice seeing this collaboration unveil so nicely. From my side, it's ready for merge. I'll get some 🍿 watching the pipeline run

**Pull request merged**